clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -498,6 +498,7 @@ void pollSelectionKeys(Set<SelectionKey> selectionKeys,
 
         if (!outOfMemory && memoryPool.availableMemory() < lowMemThreshold) {
             List<SelectionKey> temp = new ArrayList<>(selectionKeys);
+            selectionKeys.clear(); // needed for next `select` to count these keys
```

Comment:
I think I'd mention in the comment that we normally remove from `selectionKeys` during iteration, but since we are making a copy here, we need to clear it manually.

Reply:
Removed this change.